src/intersections/lineraysegmentplane.jl

```diff
@@ -0,0 +1,114 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Return appropriate type for a geometry overlapping with a `Plane`
+
+### Notes
+
+- Ideally this would be a macro, but the geometry type isn't known at parse time, so it is implemented with multiple-dispatch instead
+"""
+function planeintersectionoverlapping(::Line)
+  return OverlappingLinePlane
+end
+
+function planeintersectionoverlapping(::Ray)
+  return OverlappingRayPlane
+end
+
+function planeintersectionoverlapping(::Segment)
+  return OverlappingSegmentPlane
+end
+
+"""
+    checkparameterplaneintersection(f, l::Line, λ)
+
+Return the intersection for a `Plane` and non-parallel `Line`, `l` given a line parameter `λ`. As a line is infinite, a `CrossingLinePlane` will always be returned.
+"""
+function checkparameterplaneintersection(f, l::Line, λ)
```

Comment:
Let's rename this internal function to something short like `_intersection` and replace the docstrings by simple comments for maintainers. Docstrings aren't very useful here given that this is an internal function.